guide/phone_store/phone_store_service.bal

```diff
@@ -0,0 +1,241 @@
+
+            log:printInfo("order will be added to the order  Queue; CustomerName: '" + newOrder.customerName +
+                    "', OrderedPhone: '" + newOrder.orderedPhoneName + "';");
+
+            // Send the message to the JMS queue
+            _ = jmsProducer -> send(queueMessage);
+
+            // Construct a success message for the response
+            responseMessage = {"Message":"Your order is successfully placed. Ordered phone will be delivered soon"};
+        }
+        else {
+            // If phone is not available, construct a proper response message to notify user
+            responseMessage = {"Message":"Requested phone not available"};
+        }
+        
+        // Send response to the user
+        response.setJsonPayload(responseMessage);
+        _ = caller -> respond(response);
+    }
+    // Resource that allows users to get a list of all the available phones
+    @http:ResourceConfig {methods:["GET"], produces:["application/json"]}
+    getPhoneList(endpoint client, http:Request request) {
+        http:Response response;
+        // Send json array 'phoneInventory' as the response, which contains all the available phones
+        response.setJsonPayload(phoneInventory);
+        _ = client -> respond(response);
+    }
+}
+jms:Connection conn = new({
+        initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl: "tcp://localhost:61616"
+    });
+    
+// Initialize a JMS session on top of the created connection
+jms:Session jmsSession2 = new(conn, {
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+    
+// Initialize a queue receiver using the created session
+
+endpoint jms:QueueReceiver jmsConsumer {
+    session:jmsSession2,
+    queueName:"OrderQueue"
+};
+// JMS service that consumes messages from the JMS queue
+// Bind the created consumer to the listener service
+service<jms:Consumer> orderDeliverySystem bind jmsConsumer {
+    // Triggered whenever an order is added to the 'OrderQueue'
+    onMessage(endpoint consumer, jms:Message message) {
+        log:printInfo("New order successfilly received from the Order Queue");
+        // Retrieve the string payload using native function
+        string stringPayload = check message.getTextMessageContent();
+        log:printInfo("Order Details: " + stringPayload);
+
+        http:Request enrichedreq = backendreq;
+        var clientResponse = phone_order_delivery_serviceEP->forward("/", enrichedreq);
+        match clientResponse {
+            http:Response res => {
+                log:printInfo("Order Details have sent to phone_order_delivery_service.");
+            }
+            error err => {
+                log:printInfo("forward error..................");
+            }
+        }
+    }
+}
+endpoint http:Client phone_order_delivery_serviceEP {
```

Comment:
Snake case is not really used for variable or endpoint names.

Reply:
Also, better if this client endpoint is defined before its use. i.e., move this EP up.